src/App.js

```diff
@@ -13,15 +14,17 @@ class App extends Component {
   render() {
     return (
       <Provider store={store}>
-        
+
         <Router>
           <React.Fragment>
-            <MainNavbar/> 
-            <Route exact path="/" component={MainPage} />
-            <Route exact path="/user/:username" component={UserPanel} /> 
-            <Route exact path="/:slug" component={BlogLovWeb} />
-            <Route exact path="/:slug/addpost" component={AddPost} /> 
-            <Route path="/:category" component="" />
+            <CurrentUserProvider>
```

Comment:
Why do you use a Context API? We are using a Redux. I don't understand why we should use this when all functions from Context which are necessary in our app we have in Redux